app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/EncryptedSaver.java

```diff
@@ -0,0 +1,266 @@
+package com.nervousfish.nervousfish.modules.cryptography;
+
+
+import android.util.Base64;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Random;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.PBEParameterSpec;
+
+/**
+ * Class used for encrypting strings.
+ * Commented out a bunch of methods in case of future needs.
+ */
+public final class EncryptedSaver {
```

Comment:
Also, the methods shouldn't be static imo. That creates very tight coupling to a certain class.